src/api/Nhs.Appointments.Api/Notifications/UserRolesChangedNotifier.cs

```diff
@@ -0,0 +1,36 @@
+﻿using Microsoft.Extensions.Options;
+using Nhs.Appointments.Core;
+using Notify.Interfaces;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Nhs.Appointments.Api.Notifications;
+
+// Depends on an email template hosted in Gov Notify. See template.txt for details 
+public class UserRolesChangedNotifier(IAsyncNotificationClient notificationClient, IRolesStore rolesStore, ISiteSearchService siteService, IOptions<UserRolesChangedNotifier.Options> options) : IUserRolesChangedNotifier
+{
+    public async Task Notify(string user, string siteId, string[] rolesAdded, string[] rolesRemoved)
+    {
+        var roles = await rolesStore.GetRoles();
+        var site = await siteService.GetSiteByIdAsync(siteId);
+        var siteName = site == null ? $"Unknown site ({siteId})" : site.Name;
+        var rolesAddedNames = roles.Where(r => rolesAdded.Contains(r.Id)).Select(r => r.Name).ToArray();
```

Comment:
Unless I'm just being blind and can't see it, I don't think we're validating the roles supplied by the function invoker before this point? 
This means if roles have been added that aren't present in the response from `await rolesStores.GetRoles();` we'll silently miss them. If we had a frontend bug which supplied the wrong ids for a role, we potentially wouldn't know about it. Perhaps we could throw in this case? Or add a validation layer earlier?